pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,51 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/config"
+	"github.com/guacsec/guac/pkg/ingestor/collector/gcs"
+	"github.com/guacsec/guac/pkg/ingestor/collector/oci"
+	"github.com/guacsec/guac/pkg/ingestor/collector/pubsub"
+	"github.com/guacsec/guac/pkg/ingestor/collector/transparency"
+	"github.com/guacsec/guac/pkg/ingestor/processor"
+	"go.uber.org/zap"
+)
+
+type Collector interface {
+	RetrieveArtifacts(ctx context.Context) (map[string]*processor.Document, error)
```

Comment:
can you describe what the string in the map represents?

Reply:
Forgot to make this into a slice after making it accept `document`. Changing.